File: frontend/src/constants/unitStatus.ts
/**
 * 单元资产状态常量定义
 */

// 单元状态枚举
export enum UnitStatus {
  RENTABLE = 'RENTABLE',
  SELF_USE = 'SELF_USE',
  PUBLIC_USE = 'PUBLIC_USE',
  LEASE_BACK = 'LEASE_BACK',
  DISABLED = 'DISABLED',
  SELF_RENTAL = 'SELF_RENTAL'
}

// 单元状态显示名称映射
export const UNIT_STATUS_NAMES: Record<string, string> = {
  [UnitStatus.RENTABLE]: '可租',
  [UnitStatus.SELF_USE]: '自用',
  [UnitStatus.PUBLIC_USE]: '公用',
  [UnitStatus.LEASE_BACK]: '返租',
  [UnitStatus.DISABLED]: '停用',
  [UnitStatus.SELF_RENTAL]: '自持出租'
}

// 单元状态标签颜色映射
export const UNIT_STATUS_TAG_COLORS: Record<string, string> = {
  [UnitStatus.RENTABLE]: 'success',
  [UnitStatus.SELF_USE]: 'primary',
  [UnitStatus.PUBLIC_USE]: 'info',
  [UnitStatus.LEASE_BACK]: 'warning',
  [UnitStatus.DISABLED]: 'danger',
  [UnitStatus.SELF_RENTAL]: 'success'
}

// 单元状态选项列表（用于下拉框）
export const UNIT_STATUS_OPTIONS = [
  { label: UNIT_STATUS_NAMES[UnitStatus.RENTABLE], value: UnitStatus.RENTABLE },
  { label: UNIT_STATUS_NAMES[UnitStatus.SELF_USE], value: UnitStatus.SELF_USE },
  { label: UNIT_STATUS_NAMES[UnitStatus.PUBLIC_USE], value: UnitStatus.PUBLIC_USE },
  { label: UNIT_STATUS_NAMES[UnitStatus.LEASE_BACK], value: UnitStatus.LEASE_BACK },
  { label: UNIT_STATUS_NAMES[UnitStatus.DISABLED], value: UnitStatus.DISABLED },
  { label: UNIT_STATUS_NAMES[UnitStatus.SELF_RENTAL], value: UnitStatus.SELF_RENTAL }
]

/**
 * 获取单元状态显示名称
 */
export const getUnitStatusName = (status: string): string => {
  return UNIT_STATUS_NAMES[status] || status
}

/**
 * 获取单元状态标签颜色
 */
export const getUnitStatusTagColor = (status: string): string => {
  return UNIT_STATUS_TAG_COLORS[status] || 'info'
}

/**
 * 检查单元状态是否有效
 */
export const isValidUnitStatus = (status: string): boolean => {
  return Object.values(UnitStatus).includes(status as UnitStatus)
}

File: backend/src/main/java/com/cysz/minimal/enums/UnitStatus.java
package com.cysz.minimal.enums;

/**
 * 单元资产状态枚举
 */
public enum UnitStatus {
    
    RENTABLE("RENTABLE", "可租"),
    SELF_USE("SELF_USE", "自用"),
    PUBLIC_USE("PUBLIC_USE", "公用"),
    LEASE_BACK("LEASE_BACK", "返租"),
    DISABLED("DISABLED", "停用"),
    SELF_RENTAL("SELF_RENTAL", "自持出租");
    
    private final String code;
    private final String displayName;
    
    UnitStatus(String code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }
    
    public String getCode() {
        return code;
    }
    
    public String getDisplayName() {
        return displayName;
    }
    
    /**
     * 根据代码获取枚举
     */
    public static UnitStatus fromCode(String code) {
        if (code == null) {
            return null;
        }
        for (UnitStatus status : UnitStatus.values()) {
            if (status.getCode().equals(code)) {
                return status;
            }
        }
        throw new IllegalArgumentException("未知的单元状态代码: " + code);
    }
    
    /**
     * 根据代码获取显示名称
     */
    public static String getDisplayNameByCode(String code) {
        UnitStatus status = fromCode(code);
        return status != null ? status.getDisplayName() : code;
    }
    
    /**
     * 检查代码是否有效
     */
    public static boolean isValidCode(String code) {
        if (code == null) {
            return false;
        }
        for (UnitStatus status : UnitStatus.values()) {
            if (status.getCode().equals(code)) {
                return true;
            }
        }
        return false;
    }
}